ClosedXML/Excel/CalcEngine/Expression.cs

```diff
@@ -2,15 +2,29 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
-using System.Globalization;
 using System.Linq;
-using System.Threading;
 
 namespace ClosedXML.Excel.CalcEngine
 {
+    [Flags]
+    internal enum CoercionConvention
+    {
+        None = 0,
+        EmptyStringAsZero = 1 << 0,
+        NonEmptyStringAsZero = 1 << 1,
+        NullAsZero = 1 << 2,
+        CaseInsensitive = 1 << 3,
+
+        Default = NullAsZero,
+        AnyStringAsZero = EmptyStringAsZero | NonEmptyStringAsZero,
+        StringOrNullAsZero = AnyStringAsZero | NullAsZero
+    }
+
     internal abstract class ExpressionBase
     {
         public abstract string LastParseItem { get; }
+
+        public abstract Expression WithCoercionConvention(CoercionConvention convention);
     }
```

Comment:
I wonder, what is the purpose of this class. It is not used anywhere except from the `Expression` class, and `WithCoercionConvention` returns an instance of `Expression` and not `ExpressionBase` as one would expect. Can't we just drop it and make corresponding members of `Expression` virtual?